bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/StateDescriptionFragmentBuilder.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.types;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Builds a {@link StateDescriptionFragment} only with the relevant parts.
+ *
+ * @author Henning Treu - initial contribution and API.
+ *
+ */
+@NonNullByDefault
+public class StateDescriptionFragmentBuilder {
+
+    private @Nullable BigDecimal minimum;
+    private @Nullable BigDecimal maximum;
+    private @Nullable BigDecimal step;
+    private @Nullable String pattern;
+    private @Nullable Boolean readOnly;
+    private @Nullable List<StateOption> options;
+
+    public static StateDescriptionFragmentBuilder instance() {
+        return new StateDescriptionFragmentBuilder();
+    }
+
+    public StateDescriptionFragment build() {
+        return new StateDescriptionFragment() {
```

Comment:
Hm, I would expect that if I call `build` on a fragment and I change the fragment after that, that it does not change the previously build reference...